src/test/groovy/life/qbic/datamodel/dtos/projectmanagement/ProjectIdentifierSpec.groovy

```diff
@@ -40,4 +40,22 @@ class ProjectIdentifierSpec extends Specification{
 
     }
 
+    def "Test equals method fulfills full method contract"() {
+        when:
+        ProjectIdentifier idX = new ProjectIdentifier(new ProjectSpace(projectSpace), new ProjectCode(projectCode))
+        ProjectIdentifier idY = new ProjectIdentifier(new ProjectSpace(projectSpace), new ProjectCode(projectCode))
+        ProjectIdentifier idZ = new ProjectIdentifier(new ProjectSpace(projectSpace), new ProjectCode(projectCode))
+
+        then:
+        idX.equals(idX) == true // Reflexive
+        (idX.equals(idY) == true) && (idY.equals(idX) == true) // Symmetric
+        (idX.equals(idY) == true) && (idY.equals(idZ) == true) && (idX.equals(idZ) == true) // Transitive
+        (idX.equals(idX) == true) && (idX.equals(idX) == true) // Consistent
+        idX.equals(null) == false // Non-nullity
+
+        where:
+        projectSpace | projectCode
+        "TEST" | "QTEST"
+    }
+
```

Comment:
I think it would be better to split this into three tests for all the different properties.

Reply:
why would it be better?

Reply:
It would make the tests simpler. You wouldn't need to look into the code and read the comments to understand what is tested here.